packages/dd-trace/src/proxy.js

```diff
@@ -47,7 +51,26 @@ class Tracer extends NoopProxy {
         // dirty require for now so zero appsec code is executed unless explicitly enabled
         if (config.appsec.enabled) {
           require('./appsec').enable(config)
+        } else if (config.appsec.enabled === undefined) { // only activate remote config when conf is not set locally
+          rc.updateCapabilities(RemoteConfigCapabilities.ASM_ACTIVATION, true)
+
+          rc.on('ASM_FEATURES', (action, conf) => {
+            if (conf && conf.asm && typeof conf.asm.enabled === 'boolean') {
+              if (action === 'apply' || action === 'modify') {
+                action = conf.asm.enabled // take control
+              } else {
+                action = config.appsec.enabled // give back control to local config
+              }
```

Comment:
Seems to me like the purpose of "action" before this conditional and after are different. Wouldn't it be clearer to split that logic into separate variables?

Reply:
Yeah I have a bad abit of reusing variables, i'll clear that up

Reply:
Reusing variables with different types can sometimes cause deopts. It's best to avoid.

Reply:
Ah alright, thanks for the tip :+1: